packages/parser-css/tests/integration-tests.ts

```diff
@@ -0,0 +1,74 @@
+import * as sinon from 'sinon';
+import test from 'ava';
+import { EventEmitter2 } from 'eventemitter2';
+
+import * as postcss from 'postcss';
+import CSSParser from '../src/parser';
+
+const element = {
+    getAttribute(): string | null {
+        return null;
+    },
+    outerHTML(): Promise<string> {
+        return Promise.resolve('');
+    }
+};
+
+let sandbox: sinon.SinonSandbox;
+
+test.beforeEach((t) => {
+    t.context.element = element;
+    t.context.postcss = postcss();
+    t.context.engine = new EventEmitter2({
+        delimiter: '::',
+        maxListeners: 0,
+        wildcard: true
+    });
+
+    sandbox = sinon.createSandbox();
+    sandbox.spy(t.context.engine, 'emitAsync');
+
+    new CSSParser(t.context.engine); // eslint-disable-line
+});
+
+test.afterEach(() => {
+    sandbox.restore();
+});
+
+test.serial('If a wellformed CSS file is provided, then we should still be able to parse it and emit a parse::end::css event', async (t) => {
+    const code = '.foo { color: #fff }';
+
+/*     await t.context.engine.emitAsync('fetch::end::css', {
+        resource: 'styles.css',
+        response: {
+            body: { content: code },
+            mediaType: 'text/css'
+        }
+    }); */
+
+    const args = t.context.engine.emitAsync.args[2];
+    const [eventName, ast] = args;
+
+    t.is(eventName, 'parse::end::css');
+    t.is(ast.code, code);
+    t.is(ast.resource, 'styles.css');
+});
+
+test.serial('If a malformed CSS file is provided, then we should still be able to parse it and emit a parse::end::css event', async (t) => {
+    const code = '.foo { color: #fff } a {';
+
+/*     await t.context.engine.emitAsync('fetch::end::css', {
+        resource: 'styles.css',
+        response: {
+            body: { content: code },
+            mediaType: 'text/css'
+        }
+    }); */
+
```

Comment:
@antross I'm not sure how to do a real integration test without mocking the `fetch::end::css`. Could you guide me a bit?

Reply:
Mocking `fetch::end::css` is fine. That's consistent with what we do for similar tests in `parser-html` and `parser-javascript`. My main interest is having some coverage that actually runs `postcss` with both well-formed and malformed input which this would still do.

Reply:
👍